source/renderer/app/components/staking/stake-pools/FilterPopOver.js

```diff
@@ -0,0 +1,253 @@
+
+const messages = defineMessages({
+  resetFilter: {
+    id: 'staking.stakePools.filter.resetFilter',
+    defaultMessage: '!!!Reset Filter',
+    description: 'Reset Filter button label.',
+  },
+  retiringPools: {
+    id: 'staking.stakePools.filter.retiringPools',
+    defaultMessage: '!!!Retiring pools',
+    description: 'Retiring pools filter type.',
+  },
+  privatePools: {
+    id: 'staking.stakePools.filter.privatePools',
+    defaultMessage: '!!!Private pools',
+    description: 'Private pools filter type.',
+  },
+  poolsWithoutOffChainData: {
+    id: 'staking.stakePools.filter.poolsWithoutOffChainData',
+    defaultMessage: '!!!Pools without off-chain data',
+    description: 'Pools without off-chain data filter type.',
+  },
+  apply: {
+    id: 'staking.stakePools.filter.apply',
+    defaultMessage: '!!!Apply',
+    description: 'Filter button label.',
+  },
+});
+
+export type FilterPopOverProps = {
+  populatedFilterOptions: StakePoolFilterOptionsType,
+  onFilter: Function,
+  numberOfFilterDimensionsApplied: number,
+  isFilterDisabled: boolean,
+};
+
+@observer
+export default class FilterPopOver extends Component<FilterPopOverProps> {
+  static contextTypes = {
+    intl: intlShape.isRequired,
+  };
+
+  form: ReactToolboxMobxForm;
+  popoverTippyInstance: ElementRef<*> = createRef();
+
+  constructor(props: FilterPopOverProps, context: Object) {
+    super(props);
+
+    const {
+      populatedFilterOptions: {
+        retiringPoolsChecked,
+        privatePoolsChecked,
+        poolsWithoutOffChainDataChecked,
+      },
+    } = props;
+
+    const { intl } = context;
+
+    this.form = new ReactToolboxMobxForm({
+      fields: {
+        retiringPoolsChecked: {
+          type: 'checkbox',
+          label: intl.formatMessage(messages.retiringPools),
+          value: retiringPoolsChecked,
+        },
+        privatePoolsChecked: {
+          type: 'checkbox',
+          label: intl.formatMessage(messages.privatePools),
+          value: privatePoolsChecked,
+        },
+        poolsWithoutOffChainDataChecked: {
+          type: 'checkbox',
+          label: intl.formatMessage(messages.poolsWithoutOffChainData),
+          value: poolsWithoutOffChainDataChecked,
+        },
+      },
+    });
+  }
+
+  fillFormFields = (filterOptions: StakePoolFilterOptionsType) => {
+    const {
+      retiringPoolsChecked,
+      privatePoolsChecked,
+      poolsWithoutOffChainDataChecked,
+    } = filterOptions;
+
+    this.form.select('retiringPoolsChecked').set(retiringPoolsChecked);
+    this.form.select('privatePoolsChecked').set(privatePoolsChecked);
+    this.form
+      .select('poolsWithoutOffChainDataChecked')
+      .set(poolsWithoutOffChainDataChecked);
+  };
+
+  resetForm = () => this.fillFormFields(emptyStakePoolFilterOptions);
+
+  isFormValuesEqualTo = (comparedFilterOptions: StakePoolFilterOptionsType) => {
+    const formFieldNames = Object.keys(this.form.fields.toJSON());
```

Comment:
Is it necessary to transform fields to JSON?

Reply:
Yes this is necessary to check equality of previous form field statuses. If you remove this part, form doesn't work as expected after running app. @tomislavhoracek